src/Helpers/JsonLd.php

```diff
@@ -18,4 +29,139 @@ public static function getType($thing)
+            if(substr($methodName, 0, 3) !== "get") {
+                continue;
+            }
+
+            // Attribute name is method name without the leading "get" string,
+            // and camel-cased
+            $attrName = Str::camel(substr($methodName, 3));
+
+            // Attribute value is the result of calling $methodName on $obj
+            $attrValue = $obj->$methodName();
+
+            if(is_array($attrValue)) {
+                // If attribute value is an array,
+                // get data for serialization from each of the item.
+                foreach($attrValue as $idx => $item) {
+                    if(is_object($item)) {
+                        // Get fully qualified namespace of the item's class name
+                        $fq_classname = "\\".get_class($item);
+
+                        $item = self::prepareDataForSerialization(
+                            $item
+                        );
+                    }
+
+                    $attrValue[$idx] = $item;
+                }
+            } else if(is_object($attrValue)) {
+                // If attribute value is an object,
+                // get the data for the individual object
+
+                // Get fully qualified namespace of the item's class name
+                $fq_classname = "\\".get_class($attrValue);
+
+                switch ($fq_classname) {
+                    case "\\DateInterval":
+                        // Get interval spec string, e.g. "P1D"
+                        $attrValue = DateIntervalHelper::specString($attrValue);
+                        break;
+                    case "\\DateTime":
+                        // Get ISO 8601 date time representation,
+                        // e.g. "2019-01-01T00:00:00-08:00"
+                        $attrValue = DateTimeHelper::iso8601($attrValue);
+                        break;
+                    default:
+                        $attrValue = self::prepareDataForSerialization(
+                            $attrValue
+                        );
+                }
+            }
+
+            $data[$attrName] = $attrValue;
+        }
+
+        // Remove empty elements
+        return array_filter(
+            $data,
+            function($value) {
+                if(is_array($value) === TRUE && count($value) === 0) {
+                    // Filter out empty arrays
+                    return false;
+                }
+
+                // Filter out null values and empty strings
+                return $value !== null && $value !== "";
+            }
+        );
+    }
+
+    /**
+     * Remove all "@context" attributes from the given JSON-LD string,
+     * leaving only the first occurrence.
+     *
+     * @param string $json
+     * @return string
+     * @see https://github.com/openactive/OpenActive.NET/blob/master/OpenActive.NET/OpenActiveSerializer.cs#L113 For .NET implementation
+     */
+    public static function removeAllButFirstContext($json)
```

Comment:
What is the reason for acting on a JSON-encoded string rather than deserializing into native PHP types, acting then re-serializing? Or even better editting the data before it is encoded - AFAICT this method is used directly after json_encode so why not just change the @context entry before encoding?

Reply:
That's a good question - I've mostly reproduced what the .NET library does, which is exactly this, strip out the additional `@context` after serialization, I think mostly because the serialization happens recursively (e.g. sub-objects that need serializing call the same method), so to make things quick, replacing strings was the quick win.
I'm happy to investigate a better way if you want.

Reply:
What JSON-LD object are we passing into this? the entire response, or is it on a per json-ld document basis?

Reply:
If I understand correctly, it's the whole response right before returning from serialization.

Reply:
If it's the whole response, then the issue could be if the instances of the json-ld docs are within an array. Double checking the OA docs a moment

Reply:
Yeha, looks like RPDE is an array of items, each with a @context - it would be wrong to strip it off entirely as these are then lacking a context.

Reply:
```
{
  "next": "https://www.example.com/api/rpde/sessions?afterTimestamp=1521565719&afterId=1402CBP20150217",
  "items": [
    {
      "state": "updated",
      "kind": "SessionSeries.ScheduledSession",
      "id": "1402CBP20150217",
      "modified": 1521565719,
      "data": {
        "@context": "https://openactive.io/",
      }
   }
}
```
if there happens to be more than 1 item, I think the correct behaviour here would be to let each item have their own @context.

Reply:
Solved in https://github.com/openactive/models-php/pull/22